hasher-matcher-actioner/hmalib/matchers/matchers_base.py

```diff
@@ -183,68 +192,84 @@ def filter_match_results(self, results: t.List[IndexMatch]) -> t.List[IndexMatch
         consider extending this class and implementing your own.
         """
 
-        filtered_results_based_on_active = []
+        filtered_results = []
         for match in results:
-            match.metadata["privacy_groups"] = list(
-                filter(
-                    lambda x: get_privacy_group_matcher_active(str(x)),
-                    match.metadata.get("privacy_groups", []),
-                )
-            )
-
-            if len(match.metadata["privacy_groups"]) != 0:
-                filtered_results_based_on_active.append(match)
-
-        # Also check and see if there is a custom threshold filter
-        filtered_results_based_on_threshold = []
-        for match in filtered_results_based_on_active:
-            filtered_privacy_groups = []
-            for privacy_group in match.metadata.get("privacy_groups", []):
-                if threshold := get_optional_privacy_group_matcher_pdq_theshold(
-                    str(privacy_group)
-                ):
-                    if match.distance >= threshold:
-                        logger.debug(
-                            "Match was found at %s distance but exceeds custom theshold of %s for pg, %s",
-                            match.distance,
-                            threshold,
-                            privacy_group,
+            match.metadata = [
+                metadata_obj
+                for metadata_obj in match.metadata
+                if metadata_obj.get_source() == BANKS_SOURCE_SHORT_CODE
+                or (
+                    # If metadata obj is from threatexchange (one privacy group
```

Comment:
@BarrettOlson Special attention here. This merges changes you and I made separately.

Your changes: filter out privacy groups that have match settings threshold < the current match distance.

My changes: each metadata_obj only contains a single Indicator or BankMember.